GenderPayGap.WebUI/Views/Compare/CompareTableRows.cshtml

```diff
@@ -16,10 +16,22 @@
                         @compareReport.OrganisationName
                     </button>
                 </div>
-
+                
                 <div class="RemoveLink no-print">
-                    <a href="@Url.Action("RemoveEmployer", "Compare", new {employerIdentifier = compareReport.EncOrganisationId, returnUrl = ViewBag.ReturnUrl})">Remove <span class="visually-hidden">@compareReport.OrganisationName from comparison</span></a>
+                    @{
+                        string removeRowLink = Url.Action("RemoveEmployer", "Compare", new { employerIdentifier = compareReport.EncOrganisationId, returnUrl = ViewBag.ReturnUrl });
+                        string id = compareReport.EncOrganisationId;
+                    }
+                    <a id="remove-row-{@id}" href="@removeRowLink">Remove <span class="visually-hidden">@compareReport.OrganisationName from comparison</span></a>
+                    <script>
+                        document.getElementById("remove-row-{@id}").addEventListener("click", (e) => {
+                            e.stopPropagation()
+                            e.preventDefault()
+                            window.location.replace("@removeRowLink");
+                        })
+                    </script>

```

Comment:
Problem: Clicking 'delete' makes a post request which redirects to the same page, so clicking 'back' would just go back to the previous page which, in this case, would've been the same.

Solution: Do not add the current page to the browser history.
Note: I have to add an href attribute to `<a>` to keep the link appearance, but in practice it has 0 functionality since `window.location.replace` does all the work...